lib/ui5Framework/Sapui5MavenSnapshotResolver.js

```diff
@@ -95,8 +100,31 @@ class Sapui5MavenSnapshotResolver extends AbstractResolver {
 		}
 		const gav = metadata.gav.split(":");
 		let pkgName = metadata.npmPackageName;
-		if (!this._sources) {
+
+		// Use "npm-dist" artifact by default
+		let classifier;
+		let extension;
+		if (this._sources) {
+			// Use npm-sources artifact if sources are requested
+			classifier = "npm-sources";
+			extension = "zip";
+		} else if (this._isLegacySnapshotVersion) {
```

Comment:
might add a TODO for Tooling V4 to remove this if again?

Reply:
Done